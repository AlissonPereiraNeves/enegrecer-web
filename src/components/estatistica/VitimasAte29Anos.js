import React, { Component } from 'react';
import firebaseApp from '../../utils/firebaseUtils';
import TipoDeEstatistica from './TipoDeEstatistica';

export default class VitimasAte29Anos extends Component {
  constructor(props) {
    super(props);
    this.state = { totalDeDenunciaPorVitimasAte29Anos: '0' };
  }

  componentDidMount() {
    const ref = firebaseApp.database().ref();
    let total = 0;
    const query = ref.child('pessoasEnvolvidas').orderByKey();
    query.once('value')
      .then((snapshot) => {
        snapshot.forEach((denunciaSnapshot) => {
          const vitima = denunciaSnapshot.val().vitima;
          if (vitima && vitima.genero === 'Feminino') {
            total += 1;
          }
        });
        this.setState({ totalDeDenunciaPorVitimasMulheres: total });
      });
  }

  render() {
    return (<TipoDeEstatistica
      id="cinza"
      numero={this.state.totalDeDenunciaPorVitimasAte29Anos}
      categoria="Vitimas atÃ© 29 Anos"
    />
    );
  }
}
