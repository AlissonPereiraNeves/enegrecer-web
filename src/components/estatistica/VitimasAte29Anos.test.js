import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import VitimasAte29Anos from './VitimasAte29Anos';

describe('Vitimas ate 29 Anos', () => {
  it('deve construir a pagina corretamente ', () => {
    const vitimasAte29Anos = shallow(<VitimasAte29Anos />);
    const json = toJson(vitimasAte29Anos);
    expect(json).toMatchSnapshot();
  });
});
