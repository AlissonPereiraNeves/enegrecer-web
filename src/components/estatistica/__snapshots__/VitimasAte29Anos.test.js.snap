// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vitimas ate 29 Anos deve construir a pagina corretamente  1`] = `
<TipoDeEstatistica
  categoria="Vitimas atÃ© 29 Anos"
  id="cinza"
  numero="0"
/>
`;
